include/nbl/builtin/hlsl/tgmath/impl.hlsl

```diff
@@ -214,22 +216,27 @@ NBL_PARTIAL_REQ_TOP(concepts::FloatingPointScalar<T>)
 struct lgamma_helper<T NBL_PARTIAL_REQ_BOT(concepts::FloatingPointScalar<T>) >
 {
 	// implementation from Numerical Recipes in C, 2nd ed.
-    static T __call(T val)
-    {
-        T x, y;
-        y = x = val;
-        T tmp = x + T(5.5);
-        tmp -= (x + T(0.5)) * log_helper<T>::__call(tmp);
-
-        T ser = T(1.000000000190015);
+	static T __call(T val)
+	{
+		const T thresholds[3] = { 7e4, 1e19, 1e19 };
+		if (val > thresholds[findLSB(uint32_t(sizeof(T)))])
```

Comment:
`findLSB` isn't compile time, not sure SPIR-V and C++ optimize it out because its an intrinsic.

it would make more sense to make the `thresholds[8]` with 0s padding or make an `mpl::` thing to convert `sizeof(fundamental_scalar_t)` to a power of two size index


btw write a comment about how you arrived at these thresholds